addon/adapters/osf-adapter.js

```diff
@@ -18,97 +20,256 @@ export default DS.JSONAPIAdapter.extend(HasManyQuery.RESTAdapterMixin, DataAdapt
+            return res;
+        });
+    },
+    /**
+     * Handle removal of related resources. This differs from DELETEs in that the related
+     * record is not deleted, just dissociated from the inverse record.
+     *
+     * @param {DS.Store} store
+     * @param {DS.Snapshot} snapshot snapshot of inverse record
+     * @param {DS.Snapshot[]} removedSnapshots
+     * @param {String} relationship
+     * @param {String} url
+     * @param {Boolean} isBulk
+     **/
+    _removeRelated(store, snapshot, removedSnapshots, relationship, url, isBulk = false) {
+        return this._doRelatedRequest(store, snapshot, removedSnapshots, relationship, url, 'DELETE', isBulk).then(res => {
+            removedSnapshots.forEach(s => snapshot.record[relationship].removeCanonicalRecord(s.record));
+            return res || [];
+        });
+    },
+    /**
+     * Handle deletion of related resources
+     *
+     * @param {DS.Store} store
+     * @param {DS.Snapshot} snapshot snapshot of inverse record
+     * @param {DS.Snapshot[]} deletedSnapshots
+     * @param {String} relationship
+     * @param {String} url
+     * @param {Boolean} isBulk
+     **/
+    _deleteRelated(store, snapshot, deletedSnapshots, relationship, url, isBulk = false) { // jshint ignore:line
+        if (isBulk) {
+            return this._removeRelated(...arguments).then(() => {
+                deletedSnapshots.forEach(s => s.record.unloadRecord());
+            });
         } else {
-            return this.buildURL(...arguments);
+            return Ember.RSVP.allSettled(deletedSnapshots.map(r => r.destroyRecord()));
         }
     },
-    _serializeHasMany(serialized) {
-        if (serialized.length > 1) {
-            serialized = {
-                data: serialized.map(function(record) {
-                    var data = record.data;
-                    return data;
-                })
-            };
+    /**
+     * A helper for making _*Related requests
+     *
+     * @param {DS.Store} store
+     * @param {DS.Snapshot} snapshot snapshot of inverse record
+     * @param {DS.Snapshot[]} relatedSnapshots
+     * @param {String} relationship
+     * @param {String} url
+     * @param {String} requestMethod
+     * @param {Boolean} isBulk
+     **/
+    _doRelatedRequest(store, snapshot, relatedSnapshots, relationship, url, requestMethod, isBulk = false) {
+        var data = {};
+        var relatedMeta = snapshot.record[relationship].meta();
+        var type = singularize(relatedMeta.type);
+        var serializer = store.serializerFor(type);
+        if (relatedSnapshots.length > 1) {
+            serializer.serializeIntoHash(
+                data,
+                store.modelFor(type),
+                relatedSnapshots, {
+                    forRelationship: true,
+                    isBulk: isBulk
+                }
+            );
         } else {
-            serialized = serialized[0];
+            data.data = relatedSnapshots.map(relatedSnapshot => {
```

Comment:
Also getting same JS error here in trying to map the snapshot. `Uncaught TypeError: relatedSnapshots.map is not a function`